src/script/common/c_content.cpp

```diff
@@ -1515,6 +1554,49 @@ ToolCapabilities read_tool_capabilities(
 	return toolcap;
 }
 
+/******************************************************************************/
+WearBarParams read_wear_bar_params(
+		lua_State *L, int table)
+{
+	WearBarParams params;
+	std::basic_string<char> default_color_string;
+	if (getstringfield(L, table, "default", default_color_string))
+		parseColorString(default_color_string, params.defaultColor, false);
+	bool blend;
+	if (!getboolfield(L, table, "blend", blend))
+		blend = false;
+	params.blend = blend;
+
+	int table_values = lua_gettop(L);
+	lua_pushnil(L);
+	while(lua_next(L, table_values) != 0) {
+		// key at index -2 and value at index -1 within table_values
+		// color value entries are stored in a flat 'list' along with the default color,
+		// but in non-blend mode color entries can be distinguished by that value being a table
+		// (in blend mode, color entries can be distinguished by having a numerical key)
+		if (lua_istable(L, -1)) {
```

Comment:
because it needs to skip the `blend = false` and `default = "<color>"` entries

I'm considering changing the API to something like
```lua
wear_color = {
    -- 'constant', 'linear', 'quadratic'
    -- (nil defaults to 'constant', which is equivalent to current blend=false
    blend = "linear",
    color_stops = {
        [0.0] = "#ff0000",
        [0.5] = "#ffff00",
        [1.0] = "#00ff00",
    }
}
```

Reply:
Done